CONTRIBUTING.md

```diff
@@ -1,31 +1,368 @@
+   [Pylint](https://pylint.pycqa.org/en/latest/).
+1. Submit your PR for review.
+
+
+# New Tutorials
+
+There are three types of tutorial content that we host on
+[`pytorch.org/tutorials`](https://github.com/pytorch/tutorials):
+
+* **Interactive tutorials** are authored and submitted as Python files.
+  The build system  converts these into Jupyter notebooks and HTML. The
+  code in these tutorials is run every time they are built. To keep
+  these tutorials up and running all their package dependencies need to
+  be resolved--which makes it more challenging to maintain this type of
+  tutorial. 
+
+* **Non-interactive tutorials** are authored and submitted as
+  reStructuredText files. The build system only converts them into HTML;
+  the code in them does not run on build. These tutorials are easier to
+  create and maintain but they do not provide an interactive experience.
+  An example is the [Dynamic Quantization
+  tutorial](https://pytorch.org/tutorials/recipes/recipes/dynamic_quantization.html).
+
+* **Recipes** are tutorials that provide bite-sized, actionable
+  examples of how to use specific features, which differentiates them
+  from full-length tutorials. Recipes can be interactive or
+  non-interactive.
+
+
+# Managing data that is used by your tutorial 
+
+Your tutorial might depend on external data, such as pre-trained models,
+training data, or test data. We recommend storing this data in a
+commonly-used storage service, such as Amazon S3, and instructing your
+users to download the data at the beginning of your tutorial. 
+
+The
+[Makefile](https://github.com/pytorch/tutorials/blob/master/Makefile)
+that we use to build the tutorials contains automation that downloads
+required data files.
+
+
+# Python packages used by your tutorial
+
+We recommend that you use only mature, well-supported packages in your
+tutorial. Packages that are obscure or not well-maintained may break as
+a result of, for example, updates to Python or PyTorch or other
+packages. If your tutorial fails to build in our Continuous Integration
+(CI) system, we might contact you in order to resolve the issue.
+
+
+# Deprecation of tutorials 
+
+Under some circumstances, we might deprecate--and subsequently
+archive--a tutorial removing it from the site. For example, if the
+tutorial breaks in our CI and we are not able to resolve the issue and
+are also not able to reach you, we might archive the tutorial. In these
+situations, resolving the breaking issue would normally be sufficient to
+make the tutorial available again.
+
+Another situation in which a tutorial might be deprecated is if it
+consistently receives low ratings--or low usage--by the community. Again,
+if this occurs, we will attempt to contact you.
+
+If we identify, or suspect, that your tutorial--or a package that your
+tutorial uses--has a **security or privacy** issue, we will immediately
+take the tutorial off the site.
+
+
+# Guidance for authoring tutorials and recipes #
```

Comment:
do you need that "#" at the end of the line?

Reply:
:-P . . . No, the trailing hash marks are optional; it is just a habit that I am in when writing Markdown. They are there in all the headers that I added . . . :-P . . . I don't want to change all of them right now because I am concerned that in making a bunch of edits, I might disrupt something else in the file. The trailing hash marks are not visible when the file renders, so my preference is to merge the current version and then remove the hash marks in some future PR.